core/domain/acl_decorators_test.py

```diff
@@ -219,9 +219,10 @@ def setUp(self):
         rights_manager.publish_collection(self.owner, self.published_col_id)
 
     def test_guest_is_redirected_to_login_page(self):
-        response = self.mock_testapp.get(
-            '/mock/%s' % self.published_col_id, expect_errors=True)
-        self.assertEqual(response.status_int, 302)
+        with self.swap(self, 'testapp', self.mock_testapp):
```

Comment:
Couldn't you use self.mock_testapp directly, like before? Ditto below.

Reply:
I could, but that would be very inconsistent with other uses of `get_json` in the file. For example, [see L67, L72, L79 and so on](https://github.com/oppia/oppia/blob/develop/core/domain/acl_decorators_test.py#L67). Should those be changed too?

Reply:
@seanlip, using get_json is better I think (to stick to the pattern). We had deviated from the pattern only for 302 responses as it was causing some problems.

Reply:
Oh, sorry! I misread this and you're both correct. Please ignore my comment.